uploads.groovy

```diff
@@ -0,0 +1,144 @@
+                    cosa buildextend-aliyun \
+                           --upload \
+                           --arch=${basearch} \
+                           --build=${buildID} \
+                           --region=${primary_region} \
+                           --bucket ${bucket} \
+                           --credentials-file=\${ALIYUN_IMAGE_UPLOAD_CONFIG}
+                       """)
+                }
+            }
+        }
+        // For azure we are using two files for RHCOS secret: azure.json and azureProfile.json
+        // We need to investigate if we can have a single secret (token) to upload it,
+        // making it similar to FCOS
+        if (it == 'azure') {
+            uploaders["azure"] = {
+                tryWithCredentials([file(variable: "AZURE_IMAGE_UPLOAD_CONFIG",
+                                    credentialsId: "azure-image-upload-config")]) {
+                    def container = pipecfg.clouds.azure.container
+                    def resource_group = pipecfg.clouds.azure.resource_group
+                    def storage_account = pipecfg.clouds.azure.storage_account
+                    shwrap("""cosa buildextend-azure \
+                        --upload \
+                        --auth \${AZURE_IMAGE_UPLOAD_CONFIG}/azure.json \
+                        --build=${buildID} \
+                        --container ${container} \
+                        --profile \${AZURE_IMAGE_UPLOAD_CONFIG}/azureProfile.json \
+                        --resource-group ${resource_group} \
+                        --storage-account ${storage_account} \
+                        --force
+                     """)
+                }
+            }
+        }
+        if (it == 'gcp') {
+            uploaders["gcp"] = {
+                tryWithCredentials([file(variable: "GCP_IMAGE_UPLOAD_CONFIG",
+                                   credentialsId: "gcp-image-upload-config")]) {
+                    def create_image =  (pipecfg.clouds.gcp.create_image ? "--create-image=true": "")
+                    def description =  (pipecfg.clouds.gcp.description ? ('--description=' + pipecfg.clouds.gcp.description): "")
+                    def family =  (pipecfg.clouds.gcp.family.image_family ? '--family ' + pipecfg.clouds.gcp.family.image_family + '--deprecated': "")
+                    def acl = (pipecfg.clouds.gcp.public ? "--public" : "")
+                    def gcp_licenses = pipecfg.clouds.gcp.licenses
+                    def licenses = gcp_licenses.collect{"--license " + it.replace('STREAM', stream)}.join(" ")
+                    def today = shwrapCapture("date +%Y-%m-%d")
+                    family = family.replace("STREAM", stream)
+                    description = description.replace("BUILDID", buildID)
+                    description = description.replace("STREAM", stream)
+                    description = description.replace("BASEARCH", basearch)
+                    description = description.replace("TODAY", today)
```

Comment:
Now with https://github.com/coreos/coreos-ci-lib/pull/119, this would be something like:

```suggestion
                    description = utils.substituteStr(description, [
                        BUILDID: buildID,
                        STREAM: stream,
                        BASEARCH: basearch,
                        TODAY: shwrapCapture("date +%Y-%m-%d")
                    ])
```